### ISSUE TITLE
MySQL / MariaDB generated columns with enum type are not generated as readonly

### ISSUE THREAD

**Issue by LimmaPaulus** [jOOQ/jOOQ#13557]
### Expected behavior

JOOQ generates MySQL/MariaDB generated enum-typed columns as readonly.

### Actual behavior

Generated enum-fields are not readonly.

```java
    public final TableField<TestRecord, TestEnumCol> ENUM_COL = createField(DSL.name("enum_col"), SQLDataType.VARCHAR(12).asEnumDataType(org.jooq.generated.jooq_test.enums.TestEnumCol.class), this, "");

```

### Steps to reproduce the problem

- Default generator configuration
- DDL as follows:
```sql
create table test
(
    int_col  int,
    enum_col enum ('first_value', 'second_value') as (
                 case
                     when int_col = 1 then 'first_value'
                     else 'second_value'
                     end
                 )
)

```

### Versions

- jOOQ: 3.16.6 (commercial)
- Java: 11.0.11
- Database (include vendor): MariaDB 10.6
- OS: Manjaro Linux
- JDBC Driver (include name if inofficial driver): 2.7.3

**Comment by LimmaPaulus** [comment]
From a database perspective it would make a more sense to use varchar-type for columns like that, because now I have to write enum options twice (in column type and on "as" definition). However, I want to use Java-enum types in client code.

My production columns is mostly "state" columns similar to this:

```sql
    customer_order_line_state
        enum ('initial', 'released', 'reserved', 'picked',
            'partially_delivered', 'delivered', 'invoiced', 'cancelled')
        as (case
                when `cancelled_date` is not null
                    then 'cancelled'
                when `invoiced_quantity` =
                     `sales_quantity_as_warehouse_units` and
                     `delivered_quantity` =
                     `sales_quantity_as_warehouse_units`
                    then 'invoiced'
                when `delivered_quantity` = `sales_quantity_as_warehouse_units`
                    then 'delivered'
                when `delivered_quantity` > 0
                    then 'partially_delivered'
                when `picked_quantity` > 0
                    then 'picked'
                when `reserved_quantity` > 0
                    then 'reserved'
                when `release_date` is not null
                    then 'released'
                else 'initial' end)

```

**Comment by lukaseder** [comment]
Thanks for your report. It seems like these two properties are currently mutually exclusive. Should hopefully be an easy fix.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0 and 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13566).

Thanks again for your report!
